components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/template/APIConfigContextTestCase.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.core.template;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.carbon.apimgt.core.SampleTestObjectCreator;
+import org.wso2.carbon.apimgt.core.models.API;
+import org.wso2.carbon.apimgt.core.models.Endpoint;
+import org.wso2.carbon.apimgt.core.template.dto.TemplateBuilderDTO;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Test cases for APIConfigContext
+ *
+ */
+public class APIConfigContextTestCase {
+
+    @Test
+    public void testValidAPIConfigContext() throws APITemplateException {
+        APIConfigContext apiConfigContext = new APIConfigContext(SampleTestObjectCreator.createDefaultAPI().build(),
+                "org.test");
+        apiConfigContext.validate();
+        String actualPackageFromContext = (String) apiConfigContext.getContext().get("package");
+        Assert.assertEquals(actualPackageFromContext, "org.test");
+    }
+
+    @Test(expectedExceptions = APITemplateException.class)
+    public void testInvalidAPIConfigContext() throws APITemplateException {
+        API emptyNamedAPI = new API.APIBuilder("provider", "", "1.0.0").context("testcontext")
+                .id(UUID.randomUUID().toString()).build(); // empty name
+        APIConfigContext apiConfigContext = new APIConfigContext(emptyNamedAPI, "org.test");
+        apiConfigContext.validate();
+    }
+
+    // @Test
```

Comment:
While writing this test scenario, found out there is an issue in handling APIs starting with letters, and reported https://github.com/wso2/product-apim/issues/1470. Kept the code commented out becuase it needs to be added after that issue is fixed. 
Anyway, will remove this